foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/ecf/internal/InternalURLTransport.java

```diff
@@ -366,8 +369,9 @@ private IStatus transfer(final String name,final IRetrieveFileTransferContainerA
 					String userCredentials = ncb.getName()+ ":" + pcb.getPassword();
 					String basicAuth = "Basic " + new String(new Base64().encode(userCredentials.getBytes()));
 					headers.put("Authorization", basicAuth);
-					map.put(IRetrieveFileTransferOptions.REQUEST_HEADERS, headers);
 				}
+				headers.put("User-Agent", generateUserAgent()); //$NON-NLS-1$
+				map.put(IRetrieveFileTransferOptions.REQUEST_HEADERS, headers);
 			} catch(IOException | UnsupportedCallbackException ce) {
 			}
 		}
```

Comment:
This method is a bit long, I'd maybe split it up in smaller chunks that one can digest more easily.